algorithm/recursive/permutation

```diff
@@ -0,0 +1,32 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+// list all rearrangement of string n of distinct characters
+
+// pos is recursive depth 
```

Comment:
Since the variable "pos" will decrease when the recursive depth increase.
It is obvious that "pos" doesn't actually represent the depth of the recursive.